### DISCUSSION TITLE
SVG icons not rendering correctly.

### DISCUSSION THREAD

**Discussion by RylanYancey** [emilk/egui#4198] [Q&A]
Hi, I'm trying to build a 2D scene editor with egui and I'm having trouble with importing SVG image icons. Here is my current solution:

```rs
impl Toolbar {
    pub fn show(&mut self, ui: &mut Ui) {
        ui.horizontal(|ui| {
            ui.with_layout(Layout::left_to_right(Align::LEFT), |ui| {
               // icon for the file management menu
                let file_icon =
                    Image::new(include_image!("../assets/save.svg")).tint(Color32::LIGHT_YELLOW);

                // icon for settings menu
                let gear_icon = Image::new(include_image!("../assets/settings.svg"))
                    .tint(Color32::LIGHT_YELLOW);

                // use the icon for file management
                ui.menu_image_button(file_icon, |ui| {
                    if ui.button("Save").clicked() {
                        println!("CLICKED SAVE");
                    }
                // ...etc
```

Everything aligns correctly, but my textures turn out looking like this:
![Screenshot from 2024-03-20 10-32-20](https://github.com/emilk/egui/assets/76256852/ed9228dd-1b33-4d5a-97d8-b6ee54fc5b84)

It's blurry, not tinted yellow, it doesn't fill the space, and it shows a gray background for some reason. 

Heres what the icons look like in my ../assets/ folder. 
![Screenshot from 2024-03-20 10-35-05](https://github.com/emilk/egui/assets/76256852/85f4d292-fe66-4a51-92ba-3bab639e3efa)

And this is what they look like in the image viewer:
![Screenshot from 2024-03-20 10-35-53](https://github.com/emilk/egui/assets/76256852/8803f0e1-3d28-4904-a733-a4cdbc9b9159)

I need the icons to fit the space they are given, not show a gray background, be sharp, and be tinted a light yellow. How can I make sure that happens?

**Comment by YgorSouza** [comment]
To remove the button's padding and background, you can change the UI's style.

```rs
let style = ui.style_mut();
style.spacing.button_padding = [0.0; 2].into();
style.visuals.widgets.inactive.weak_bg_fill = Color32::TRANSPARENT;
style.visuals.widgets.hovered.weak_bg_fill = Color32::TRANSPARENT;
style.visuals.widgets.active.weak_bg_fill = Color32::TRANSPARENT;
```

About the `tint` method, it multiplies the colors in the image by the tint, so it won't work on black images since it is multiplying by 0. The original image has to be white for it to work properly.

As for the image being blurry, see #3501. Basically you can copy the default texture loader and make one that loads the image at 2x or 3x the size, so it looks better (at the expense of using more RAM).
